src/utils/is.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Is.js
+ *
+ * Object type checks.
+ */
+
+const NAME = Symbol.toStringTag;
+
+/**
+ * Check if `obj` is a URLSearchParams object
+ * ref: https://github.com/bitinn/node-fetch/issues/296#issuecomment-307598143
+ *
+ * @param  {*} obj
+ * @return {boolean}
+ */
+export function isURLSearchParams(obj) {
+	return (
+		typeof obj === 'object' &&
+        typeof obj.append === 'function' &&
+        typeof obj.delete === 'function' &&
+        typeof obj.get === 'function' &&
+        typeof obj.getAll === 'function' &&
+        typeof obj.has === 'function' &&
+        typeof obj.set === 'function' &&
+        typeof obj.sort === 'function' &&
+        obj[NAME] === 'URLSearchParams'
+	);
+}
+
+/**
+ * Check if `obj` is a W3C `Blob` object (which `File` inherits from)
+ *
+ * @param  {*} obj
+ * @return {boolean}
+ */
+export function isBlob(obj) {
+	return (
+		typeof obj === 'object' &&
+        typeof obj.arrayBuffer === 'function' &&
+        typeof obj.type === 'string' &&
+        typeof obj.stream === 'function' &&
+        typeof obj.constructor === 'function' &&
+        typeof obj.constructor.name === 'string' &&
+        /^(Blob|File)$/.test(obj[NAME])
+	);
```

Comment:
> drop constructor.name in favor of toStringTag in v3 release where applicable.

can we use symbol here instead?

Reply:
It turns out the check after `constructor.name === 'string'` would indirectly also test for a string so I've removed the extraneous check.